src/librustc/middle/cstore.rs

```diff
@@ -67,6 +67,8 @@ pub struct CrateSource {
 
 #[derive(RustcEncodable, RustcDecodable, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Debug)]
 pub enum DepKind {
+    /// A dependency that is only used for its macros, none of which are visible from other crates.
+    UnexportedMacrosOnly,
```

Comment:
Oh, I see -- monopolized functions from extern crates might need to access non-visible functions. I don't think this is relevant to macros though, since expansion finishes before monopolization.